app/models/hyrax/collection_type.rb

```diff
@@ -109,5 +109,19 @@ def ensure_no_collections
         errors[:base] << I18n.t('hyrax.admin.collection_types.error_not_empty')
         throw :abort
       end
+
+      def ensure_no_settings_changes
+        return true unless collections?
+        return true unless collection_type_properties_changed?
+        errors[:base] << I18n.t('hyrax.admin.collection_types.error_settings_changes')
+        throw :abort
+      end
+
+      def collection_type_properties_changed?
+        ['nestable', 'discoverable', 'sharable', 'allow_multiple_membership', 'require_membership', 'assigns_workflow', 'assigns_visibility'].any? do |key|
```

Comment:
I think a slightly simpler version of this would be using the intersection instead of any?:
(property_keys & changes.keys).present?